### ISSUE TITLE
Add support in `LoadContext` for running preprocessing up through demodulation

### ISSUE THREAD

**Issue by tskisner** [simonsobs/sotodlib#1008]
If we are running a preprocessing pipeline during load, and that pipeline config has demodulation enabled, then create demodulated toast observation containers rather than the usual ones.  The mapping between axismanager and toast detector naming conventions should be extracted to helper functions callable by the work needed to address #1007 .

**Comment by msilvafe** [comment]
Are you just planning to scan the preprocess config file or loaded pipeline class attributes for this?

**Comment by msilvafe** [comment]
I don't think so! You can just look for the config pipeline element with `name == 'demodulate' ` https://github.com/simonsobs/sotodlib/blob/f33fdd8306bcc0c65a64b33f6439ddae8f926db5/sotodlib/preprocess/processes.py#L688
